### PR TITLE
Handle record types in typeDescToWords

### LINKED ISSUE #643: Fix typeDescToWords to handle record types
They are currently getting serialized as `map<never>`.

### PR DESCRIPTION
**Pull request by ushirask**
Fixes #643

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: compiler/modules/front.syntax/astString.bal [hunk 1]
```diff
@@ -247,6 +247,18 @@ function typeDescToWords(Word[] w, TypeDesc td, boolean|BinaryTypeOp wrap = fals
...
+            w.push("record", "{|");
+            boolean firstInBlock = true;
+            foreach var f in td.fields {
+                w.push(<Word>(firstInBlock ? LF_INDENT : LF));
+                firstInBlock = false;
+                typeDescToWords(w, f.typeDesc);
+                w.push(f.name, CLING, ";");
+            }
+            w.push(<Word>(firstInBlock ? LF : LF_OUTDENT), "|}");
+            return;
```
Comment (jclark) [inline compiler/modules/front.syntax/astString.bal]: Please use `else` here.

Also handle the td.rest using `...`.

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/modules/front.syntax/astString.bal [hunk 2]
```diff
@@ -247,16 +247,33 @@ function typeDescToWords(Word[] w, TypeDesc td, boolean|BinaryTypeOp wrap = fals
...
+        if td.fields.length() > 0 {
+            w.push("record", "{|");
+            boolean firstInBlock = true;
+            foreach var f in td.fields {
+                w.push(<Word>(firstInBlock ? LF_INDENT : LF));
+                firstInBlock = false;
+                typeDescToWords(w, f.typeDesc);
+                w.push(f.name, CLING, ";");
+            }
+            if rest != () {
```
Comment (jclark) [inline compiler/modules/front.syntax/astString.bal:260]: I don't think the firstInBlock logic is working right with rest. Are we properly handling the case where there is a `rest` field but no `members`?

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/modules/front.syntax/astString.bal (hunk 2 above)
Comment (ushirask) [reply compiler/modules/front.syntax/astString.bal:260]: Can we differentiate between `type M map<int>` and
```
type R record {|
    int ...;
|}
```
in MappingTypeDesc ?

Comment (manuranga) [reply compiler/modules/front.syntax/astString.bal:260]: No. Those are two ways of writing the same thing. We should pick `map<int>` as the canonical form.

Comment (ushirask) [reply compiler/modules/front.syntax/astString.bal:260]: in the current implementation second form will also be serialized as `map<int>`

**Review by jclark — APPROVED** [review]

**Pull request merged**